globus_cli/commands/logout.py

```diff
@@ -0,0 +1,112 @@
+import click
+import globus_sdk
+
+from globus_cli.safeio import safeprint
+from globus_cli.parsing import common_options
+from globus_cli.config import (
+    AUTH_RT_OPTNAME, TRANSFER_RT_OPTNAME,
+    AUTH_AT_OPTNAME, TRANSFER_AT_OPTNAME,
+    AUTH_AT_EXPIRES_OPTNAME, TRANSFER_AT_EXPIRES_OPTNAME,
+    WHOAMI_ID_OPTNAME, WHOAMI_USERNAME_OPTNAME,
+    WHOAMI_EMAIL_OPTNAME, WHOAMI_NAME_OPTNAME,
+    internal_auth_client, remove_option, lookup_option)
+
+
+_RESCIND_HELP = """\
+Rescinding Consents
+-------------------
+The logout command only revokes tokens that it can see in its storage.
+If you are concerned that logout may have failed to revoke a token,
+you may want to manually rescind the Globus CLI consent on the
+Manage Consents Page:
+    https://auth.globus.org/consents
+
+"""
+
+
+_LOGOUT_EPILOG = """\
+\nYou are now successfully logged out of the Globus CLI.
+Before attempting any further CLI commands, you will have to login again using
+
+  globus login
+
+"""
+
+
+@click.command('logout',
+               short_help='Logout of the Globus CLI',
+               help=('Logout of the Globus CLI. '
+                     'Removes your Globus tokens from local storage, '
+                     'and revokes them so that they cannot be used anymore'))
+@common_options(no_format_option=True, no_map_http_status_option=True)
+@click.option('--yes', help='Automatically say "Yes" to all prompts',
+              is_flag=True, default=False)
+def logout_command(yes):
+    def _confirm(prompt, default=False):
+        """
+        Handy confirmation prompt that respects --yes and exits if confirmation
+        fails.
+        """
+        if yes:
+            return
+        confirmed = click.confirm(prompt, default=default)
+        if not confirmed:
+            click.get_current_context().exit(1)
+
+    # prompt
+    _confirm("Are you sure you want to logout?")
+
+    # check for username -- if not set, probably not logged in
+    username = lookup_option(WHOAMI_USERNAME_OPTNAME)
+    if not username:
+        _confirm(("Your username is not set. You may not be logged in. "
+                  "Would you like to try to logout anyway?"), default=True)
```

Comment:
I didn't know about `confirmation_option`.
I'm fine with dropping this and instead printing a message.

I think we should say something since we can't print a username.
It also makes some sense of the numerous warnings you get about missing tokens if we say "We think you are not logged in..."